src/mxarray.jl

```diff
@@ -406,13 +406,13 @@ end
 # char arrays and string
 
 function mxarray(s::String)
-    utf16string = transcode(UInt16, s)
+    utf16string = transcode(mxChar, s)
```

Comment:
Why? `transcode` is a julia function and not a MATLAB one so I don't see why this is better.